File: app/src/test/java/io/wyki/aggro/api/AggregationsApiImplTest.java
package io.wyki.aggro.api;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.quarkus.test.junit.QuarkusTest;
import io.wyki.aggro.app.generated.model.AggregationType;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;
import java.util.Arrays;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;

@QuarkusTest
@Tag("integration")
public class AggregationsApiImplTest {
    private final ObjectMapper om;

    @Inject
    public AggregationsApiImplTest(ObjectMapper om) {
        this.om = om;
    }

    @Test
    public void testHelloEndpoint() throws Exception {
        var expected = om.writeValueAsString(Arrays.asList(new AggregationType()
                .name("AVERAGE")
                .description("Aggregate values on an average basis")));
        given().when().get("/aggregations")
                .then().statusCode(200).body(is(expected));
    }
}

File: app/src/test/kotlin/io/wyki/aggro/api/AggregationsApiImplTest.kt
package io.wyki.aggro.api

/* ktlint-disable import-ordering */
import com.fasterxml.jackson.databind.ObjectMapper
import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured
import org.hamcrest.Matchers.hasItems
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import javax.enterprise.inject.Default
import javax.inject.Inject

@QuarkusTest
@Tag("integration")
class AggregationsApiImplTest {

    @Inject
    @field: Default
    private lateinit var om: ObjectMapper

    @Test
    @Throws(Exception::class)
    fun testHelloEndpoint() {
        RestAssured.given().`when`()["/aggregations"]
            .then().statusCode(200).body(
                "name", hasItems("AVERAGE", "MINIMUM", "MAXIMUM")
            )
    }
}
